### ISSUE TITLE
Calculation with money fields leads to inconsistent `decimal_places`

### ISSUE THREAD

**Issue by wearebasti** [django-money/django-money#522]
Related to #521 I realized that the decimal-places are ignored when calculations are done:

```python

In [45]: one = Money(2.000000005, "USD", decimal_places=10)

In [46]: two = Money(1.0000, "USD", decimal_places=4)

In [47]: three=one+two

In [48]: three
Out[48]: <Money: 3.000000005 USD>

In [49]: three.decimal_places
Out[49]: 2
```

As far as I can see the decimal places are always set back to the `DEFAULT_DECIMAL_PLACES` from the settings. This seems inconsistent.

**Comment by benjaoming** [comment]
Agreed - a fix for this would be most welcome :+1:

**Comment by kymair** [comment]
For division,  result here is `money.amount` which is a `Decimal` instead of `Money`. 
And it doesn't have `decimal_places`attribute. 

```
In [15]: a.aud_price
Out[15]: <Money: 16.80 AUD>

In [16]: a.last_aud_price
Out[16]: <Money: 17.00 AUD>

In [17]: a.aud_price / a.last_aud_price
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-17-bb0b061b2b62> in <module>
----> 1 a.aud_price / a.last_aud_price

~/.cache/pypoetry/virtualenvs/product-HwerkFS7-py3.7/lib/python3.7/site-packages/djmoney/money.py in __truediv__(self, other)
     59             return other.__rtruediv__(self)
     60         result = super().__truediv__(other)
---> 61         result.decimal_places = self._fix_decimal_places(other)
     62         return result
     63

AttributeError: 'decimal.Decimal' object has no attribute 'decimal_places'
```

**Comment by wearebasti** [comment]
Thanks for reporting this @kymair ! I guess this should be extracted in its own issue @benjaoming? (despite being caused by the linked PR #554 )

One obvious fix would be to check for the type of the result of the `super().__truediv__(other)` call as I of course money divided by money does not return another money-object, but a unit-less decimal...

**Comment by niklasb** [comment]
@wearebasti I made a quick patch along the lines of what you described

**Comment by wearebasti** [comment]
Haha @niklasb  you beat me to it, was just on it too :)

While looking into it I found another issue in regards of divisions involving decimals: #586

**Comment by niklasb** [comment]
@wearebasti weird timing coincidence, feel free to suggest a different PR that addresses both. we can close mine

**Comment by wearebasti** [comment]
> @wearebasti weird timing coincidence, feel free to suggest a different PR that addresses both. we can close mine

no no, all good :) It's two issues I'd argue for keeping yours and addressing the other one in a separate PR -- still needs to be decided by one of the maintainers if it should be fixed or be relayed upon the money library.

**Comment by Stranger6667** [comment]
Hi all! I assume that this issue is fixed via #554 and can be closed :) Let me know if I missed something.
The latest reported issue was also fixed (#585 #588 )
